packages/core/src/CardanoNode/util/bufferChainSyncEvent.ts

```diff
@@ -0,0 +1,50 @@
+import { Observable } from 'rxjs';
+import { RequestNext } from '../types';
+
+export interface WithRequestNext {
+  requestNext: RequestNext;
+}
+
+export const bufferChainSyncEvent =
+  <T extends WithRequestNext>(length: number) =>
+  (source$: Observable<WithRequestNext>) =>
+    new Observable<T>((subscriber) => {
+      const buffer: WithRequestNext[] = [];
+      let lastPull: RequestNext | undefined;
+      let unsubscribed = false;
+      let subscriberReady = true;
+
+      const next = () => {
+        if (unsubscribed) return;
+
+        if (buffer.length < length && lastPull) {
+          lastPull();
+          lastPull = undefined;
+        }
+
+        if (buffer.length === 0 || !subscriberReady) return;
```

Comment:
I may be wrong but if I remember correctly observable chain sync client emits 1 event initially on it's own, without calling requestNext. This may be why it's not projecting.